### ISSUE TITLE
Permission issue view other user info

### ISSUE THREAD

**Issue by Ulminia** [pods-framework/pods#5086]
**Describe the bug**
we have a custom pod called courses
users add courses to them selfs and that work but when they go to the pods_manage_course page all posts for all users in that cpt are listed not just theres

**To Reproduce**
create a pod make sure no users have access to "other" pods users 
add posts one 2 or more accounts
all posts show for all users

**Expected behavior**
when a users licks on there courses link they only see there courses

**Screenshots**
https://prnt.sc/kd04mx

## Pods Version
2.7.8

## WordPress Environment

<details>
WordPress Version: 4.9.7

PHP Version: 7.0.30-1+ubuntu14.04.1+deb.sury.org+1

MySQL Version: 5.6.40

Server Software: nginx/1.12.1

Your User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Safari/537.36

Session Save Path: /var/lib/php/sessions

Session Save Path Exists: Yes

Session Save Path Writeable: Yes

Session Max Lifetime: 1440

Opcode Cache:

Apc: No
Memcached: No
OPcache: Yes
Redis: No
Object Cache:

APC: No
APCu: No
Memcache: Yes
Memcached: Yes
Redis: No
WPDB Prefix: wp_

WP Multisite Mode: No

WP Memory Limit: 128M

Pods Network-Wide Activated: No

Pods Install Location: /home/s4/html/wp-content/plugins/pods/

Pods Tableless Mode Activated: No

Pods Light Mode Activated: No

Currently Active Theme: Intuition

Currently Active Plugins:

Advanced Custom Fields Multilingual: 0.7
Akismet Anti-Spam: 4.0.8
BackUpWordPress: 3.6.4
BackWPup: 3.6.0
Black Studio TinyMCE Widget: 2.6.2
CPO Content Types: 1.1.0
Database Browser: 1.4
Display PHP Version: 1.4
Events Manager: 5.9.5
Events Manager and WPML Compatibility: 1.2
Exports and Reports: 0.7.4
Gravity Forms: 2.3.3
Gravity Forms Multilingual: 1.3.17
Gravity Forms User Registration Add-On: 3.9
Loco Translate: 2.1.4
Menu Image: 2.9.1
NB Area directory directory: 1.0.0
Page Links To: 3.0.1
PHP code snippets (Insert PHP): 2.0.6
Pods - Custom Content Types and Fields: 2.7.8
Real-Time Find and Replace: 3.9
s2Member Framework: 170722
Shareaholic | share buttons, analytics, related posts: 8.8.0
Show Current Template: 0.3.0
SiteOrigin Widgets Bundle: 1.12.1
Username Changer: 3.1.3
Wordfence Security: 7.1.9
WPBakery Page Builder: 5.4.7
WP Clips: 2.1.1
WP Downgrade | Specific Core Version: 1.1.4
WP Edit: 4.0.3
WPML All Import: 2.0.6
WPML CMS Nav: 1.4.23
WPML Media: 2.3.5
WPML Multilingual CMS: 4.0.5
WPML Sticky Links: 1.4.4
WPML String Translation: 2.8.5
WPML Translation Management: 2.6.5
WPML XLIFF: 0.9.1
Yoast SEO: 7.9
</details>

**Comment by sc0ttkclark** [comment]
* Is this an Advanced Content Type?
* Do you have an “author” field set?
* Do you have the role set with the correct capabilities for that user to be limited?

**Comment by Ulminia** [comment]
Is this an Advanced Content Type?
yes
Do you have an “author” field set?
yes
Do you have the role set with the correct capabilities for that user to be limited?
yes

Pods Capabilities - course

access_s2member_level1
pods_add_course
edit_pods_pods
publish_pods_pods
pods_delete_course
read_pods_page
pods_edit_course
pods_export_course
read

are all checked

**Comment by sc0ttkclark** [comment]
Thanks, that helps. Could you provide an export of your “Course” pod package so we can test on our side?

**Comment by Ulminia** [comment]
here it is
<details>
{"meta":{"version":"2.7.8","build":1533042419},"pods":{"758":{"id":758,"name":"course","label":"Courses","description":"","type":"pod","storage":"table","object":"","alias":"","fields":{"course_name":{"id":1437,"name":"course_name","label":"Course Name","description":"","help":"","class":"","type":"pick","weight":0,"pick_object":"custom-simple","pick_val":"","sister_id":"","pick_custom":"Professional Practice \nAppraising in NB\nAnnual Education Seminar \nAnnual General Meeting\nCNAREA 6.5 \nCNAREA 5.0A\nCMHC\nAssessor School\nCarry Over Credits\nOther\nUBC","required":"1","unique":"0","pick_format_type":"single","pick_format_single":"dropdown","pick_format_multi":"checkbox","pick_taggable":"0","pick_limit":"0","pick_allow_html":"0","pick_user_role":[],"admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"pick_post_status":[],"pick_allow_add_new":"1","pick_show_icon":"1","pick_show_edit_link":"1","pick_show_view_link":"1","pick_select_text":"","pick_table_id":"","pick_table_index":"","pick_display":"","pick_where":"","pick_orderby":"","pick_groupby":""},"name":{"id":759,"name":"name","label":"Course Name (old)","description":"","help":"","class":"","type":"text","weight":1,"pick_object":"","pick_val":"","sister_id":"","required":"0","unique":"0","text_repeatable":"0","text_allowed_html_tags":"strong em a ul ol li b i","text_max_length":"255","text_allow_shortcode":"0","text_allow_html":"0","admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"pick_allow_html":"0","default_value_parameter":"{@course_name}","_wpml_location_migration_done":"1","text_placeholder":""},"course_description":{"id":764,"name":"course_description","label":"Course Description","description":"","help":"","class":"","type":"wysiwyg","weight":2,"pick_object":"","pick_val":"","sister_id":"","required":"1","unique":"0","wysiwyg_editor":"tinymce","wysiwyg_media_buttons":"1","wysiwyg_oembed":"0","wysiwyg_wptexturize":"1","wysiwyg_convert_chars":"1","wysiwyg_wpautop":"1","wysiwyg_allow_shortcode":"0","admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"wysiwyg_repeatable":"0","_wpml_location_migration_done":"1","wysiwyg_allowed_html_tags":""},"course_date":{"id":765,"name":"course_date","label":"Course Date","description":"","help":"","class":"","type":"date","weight":3,"pick_object":"","pick_val":"","sister_id":"","required":"1","unique":"0","date_format":"mdy","date_allow_empty":"1","date_html5":"0","admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"date_repeatable":"0","_wpml_location_migration_done":"1","date_type":"format","date_format_custom":"","date_format_custom_js":""},"learning_type":{"id":766,"name":"learning_type","label":"Learning Type","description":"","help":"","class":"","type":"pick","weight":4,"pick_object":"custom-simple","pick_val":"","sister_id":"","pick_custom":"Professional Practice Seminar\nRelated Course\nRelated Seminar\nRelated Workshop\nCarry Over Credits","required":"1","unique":"0","pick_format_type":"single","pick_format_single":"dropdown","pick_format_multi":"checkbox","pick_limit":"0","pick_user_role":[],"admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"pick_taggable":"0","pick_allow_html":"0","pick_post_status":"publish","_wpml_location_migration_done":"1","pick_allow_add_new":"1","pick_show_icon":"1","pick_show_edit_link":"1","pick_show_view_link":"1","pick_select_text":"","pick_table_id":"","pick_table_index":"","pick_display":"","pick_where":"","pick_orderby":"","pick_groupby":""},"credits":{"id":767,"name":"credits","label":"Credits","description":"","help":"","class":"","type":"number","weight":5,"pick_object":"","pick_val":"","sister_id":"","required":"1","unique":"0","number_format_type":"number","number_format":"9,999.99","number_decimals":"2","number_step":"1","number_min":"0","number_max":"100","number_max_length":"12","admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"number_repeatable":"0","_wpml_location_migration_done":"1","number_format_soft":"0","number_placeholder":""},"created":{"id":760,"name":"created","label":"Date Created","description":"","help":"","class":"","type":"datetime","weight":6,"pick_object":"","pick_val":"","sister_id":"","datetime_format":"ymd_slash","datetime_time_type":"12","datetime_time_format":"h_mm_ss_A","required":"0","unique":"0","datetime_repeatable":"0","datetime_time_format_24":"hh_mm","datetime_allow_empty":"1","datetime_html5":"0","admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"1","read_only":"0","roles_allowed":["administrator"],"_wpml_location_migration_done":"1","datetime_type":"format","datetime_format_custom":"","datetime_format_custom_js":"","datetime_time_format_custom":"","datetime_time_format_custom_js":""},"modified":{"id":761,"name":"modified","label":"Date Modified","description":"","help":"","class":"","type":"datetime","weight":7,"pick_object":"","pick_val":"","sister_id":"","datetime_format":"ymd_slash","datetime_time_type":"12","datetime_time_format":"h_mm_ss_A","required":"0","unique":"0","datetime_repeatable":"0","datetime_time_format_24":"hh_mm","datetime_allow_empty":"1","datetime_html5":"0","admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"1","read_only":"0","roles_allowed":["administrator"],"_wpml_location_migration_done":"1","datetime_type":"format","datetime_format_custom":"","datetime_format_custom_js":"","datetime_time_format_custom":"","datetime_time_format_custom_js":""},"author":{"id":762,"name":"author","label":"Member","description":"","help":"","class":"","type":"pick","weight":8,"pick_object":"user","pick_val":"","sister_id":"","pick_format_type":"single","pick_format_single":"dropdown","default_value":"{@user.ID}","required":"1","unique":"0","pick_format_multi":"checkbox","pick_limit":"0","pick_user_role":[],"admin_only":"1","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"pick_taggable":"0","pick_allow_html":"0","pick_display":"{@display_name}","pick_post_status":[],"_wpml_location_migration_done":"1","pick_allow_add_new":"1","pick_show_icon":"1","pick_show_edit_link":"1","pick_show_view_link":"1","pick_select_text":"","pick_table_id":"","pick_table_index":"","pick_where":"","pick_orderby":"","pick_groupby":""},"permalink":{"id":763,"name":"permalink","label":"Permalink","description":"Leave blank to auto-generate from Name","help":"","class":"","type":"slug","weight":9,"pick_object":"","pick_val":"","sister_id":"","required":"0","unique":"0","pick_allow_html":"0","admin_only":"1","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"_wpml_location_migration_done":"1","slug_placeholder":""}},"show_in_menu":"1","label_singular":"Course","pod_index":"name","ui_style":"post_type","ui_actions_enabled":{"0":"add","1":"edit","2":"duplicate","3":"delete","5":"export"},"ui_reorder_field":"menu_order","ui_fields_manage":{"0":"course_name","1":"name","2":"course_description","3":"course_date","4":"learning_type","5":"credits","7":"modified","8":"author"},"ui_filters":{"0":"course_name","1":"name","3":"course_date","4":"learning_type","5":"credits","6":"created","8":"author"},"hierarchical":"0","_wpml_location_migration_done":"1"}},"templates":{"Courses":{"id":1431,"name":"Courses","code":"<tr><td>{@name}<\/td><td>{@learning_type}<\/td><td>{@course_date}<\/td><\/tr>\r\n","type":"template","slug":"courses","options":{"_edit_lock":"1513954291:1","_edit_last":"1","_vc_post_settings":"a:2:{s:7:\"vc_grid\";a:0:{}s:10:\"vc_grid_id\";a:0:{}}","admin_only":"","restrict_capability":"","capability_allowed":"","pod_reference":"a:1:{s:3:\"pod\";s:6:\"course\";}"}}}}
</details>

**Comment by Ulminia** [comment]
any updates on this yet?

**Comment by jimtrue** [comment]
@Ulminia Please also provide the additional issues you're having that you mentioned on #support in Slack Chat.

**Comment by Ulminia** [comment]
this is the main issue there are no other issues at this time accept for the fact that users go to there course list they see all users courses and they should only see there own

**Comment by quasel** [comment]
@sc0ttkclark did you test that?

**Comment by Ulminia** [comment]
a side not version 2.6.11 does not have this issue or any of my previous issues i have reverted my client to that version to have her site working as she was ferrous at how long it was taking

**Comment by jimtrue** [comment]
@Ulminia Unfortunately, we can only direct resources to the highest most impacted issues, whether your client understands that or not; if your workaround was to downgrade, then that's a reasonable workaround. 

This is the nature of Open Source software; we can't expend all of our resources for one edge case when no one else is reporting any issues of this nature. Needs of the many outweigh the needs of the few or the one. It's also why we rely so heavily on the developers who work with our code on their edge cases to see if they're able to provide troubleshooting assistance and simplify the conditions of the test case to make it easier to find the root cause.

**Comment by adleviton** [comment]
I think I have a similar issue using an advanced content type. Users can see everyone else's records even though `edit_others_whatever` and `delete_others_whatever` are unchecked. The first time I see this is in v2.7.7.

![pods-issue](https://user-images.githubusercontent.com/2176348/47278213-1f7e1480-d595-11e8-9a5f-12891e395d28.jpg)

**Comment by adleviton** [comment]
Update: I just tested this and it's not an issue before **v2.7.7**.

**Comment by sc0ttkclark** [comment]
At line 4931 in classes/PodsUI.php, if you change:

```
if ( ! in_array( 'edit', $this->actions_disabled ) && ( current_user_can( 'pods_edit_' . $this->pod->pod ) || current_user_can( 'pods_edit_others_' . $this->pod->pod ) ) ) {
```

to:

```
if ( ! in_array( 'edit', $this->actions_disabled ) && current_user_can( 'pods_edit_others_' . $this->pod->pod ) ) {
```

That should resolve the issue for you. Let me know if it does and we can get a fix for this in the next release.

cc: @adleviton

**Comment by adleviton** [comment]
@sc0ttkclark 

I updated to 2.7.9, and also fixed the IF statement right after 4931 as shown below. This is working on my end, thank you!

          // if ( ! in_array( 'edit', $this->actions_disabled ) && ( current_user_can( 'pods_edit_' . $this->pod->pod ) || current_user_can( 'pods_edit_others_' . $this->pod->pod ) ) ) {
          if ( ! in_array( 'edit', $this->actions_disabled ) && current_user_can( 'pods_edit_others_' . $this->pod->pod ) ) {
            $restricted = false;
          // } elseif ( ! in_array( 'delete', $this->actions_disabled ) && ( current_user_can( 'pods_delete_' . $this->pod->pod ) || current_user_can( 'pods_delete_others_' . $this->pod->pod ) ) ) {
          } elseif ( ! in_array( 'delete', $this->actions_disabled ) && current_user_can( 'pods_delete_others_' . $this->pod->pod ) ) {
            $restricted = false;
          } elseif ( current_user_can( 'pods_' . $action . '_' . $this->pod->pod ) || current_user_can( 'pods_' . $action . '_others_' . $this->pod->pod ) ) {
            $restricted = false;
	}

**Comment by JoryHogeveen** [comment]
Fixed since: #5504
